server/src/main/java/org/elasticsearch/cluster/coordination/FollowersChecker.java

```diff
@@ -160,8 +162,12 @@ public void updateFastResponseState(final long term, final Mode mode) {
     }
 
     private void handleFollowerCheck(FollowerCheckRequest request, TransportChannel transportChannel) throws IOException {
-        FastResponseState responder = this.fastResponseState;
+        if (nodeHealthService.getHealth() == NodeHealthService.Status.UNHEALTHY) {
+            logger.error("Rejecting health check request {} as all data paths are not writable", request);
```

Comment:
An error here isn't helpful, we should be logging the failure elsewhere so this will simply result in confusion.
```suggestion
            logger.debug("Rejecting health check request {} as all data paths are not writable", request);
```


Also, we have this generic `NodeHealthService` but the log message is very specific: `all data paths are not writable`. Maybe the `NodeHealthService` should describe the problem rather than returning a simple boolean. Possibly makes sense to construct the exception within the `NodeHealthService`.

Reply:
Do you think it makes sense to change the interface to below and throwing a checked exception to the consumers
```
public interface NodeHealthService {
   void ensureHealthy() throws HealthCheckFailureException;
```

Reply:
`HealthCheckFailureException ` could either be a generic exception or `FsHealthCheckFailureException` if needed can sub-class it. Let me know what you think

Reply:
I think its better to add more info along with status since we are not actually performing health check but returning pre-computed results. I have also changed `FsHealthCheckFailureException` to `HealthCheckFailureException` to maintain the generic interface of `NodeHealthService`
```
@FunctionalInterface
public interface NodeHealthService {

    StatusInfo getHealth();

    enum Status { HEALTHY, UNHEALTHY }

    class StatusInfo {
        private Status status;
        private String info;

        public StatusInfo(Status status, String info) {
            this.status = status;
            this.info = info;
        }

        public String getInfo() {
            return info;
        }

        public Status getStatus() {
            return status;
        }
    }
}
```